c2rust-analyze/src/borrowck/type_check.rs

```diff
@@ -348,8 +348,7 @@ impl<'tcx> TypeChecker<'tcx, '_> {
             StatementKind::Assign(ref x) => {
                 let (pl, ref rv) = **x;
 
-                if matches!(rv, Rvalue::Cast(_, Operand::Copy(p) | Operand::Move(p), _) if self.special_casts.contains_key(&p))
-                {
+                if self.special_casts.is_special(&pl, &rv) {
```

Comment:
`is_special` checks `self.0.contains_key(lhs)`, so `self.special_casts.contains_key(pl)`, which wasn't checked previously here.  This seems more correct, but I just want to check to make sure.

Reply:
this is the behavior I want, albeit unintended